### PR TITLE
Fix resolution bug

### PR DESCRIPTION
**Pull request by mkholjuraev**
This fixes the invisible recommendations bug on prod-stable. The issue was due to some of the recommendations had resultion: null in the api response and all [lines](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/SmartComponents/SystemAdvisor/SystemAdvisor.js#L195) in which resolution is used to access a property caused type error. With this commit, all lines dealing with resolution are modified to handle resolution: null nicely.

To test:

- There is a temp commit with the API result from the customer. 
- Please import results object from **apiResponseInTheCase** file.
- provide the results object to [activeRuleFirst](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/SmartComponents/SystemAdvisor/SystemAdvisor.js#L618) function so that the api mock data is used to build the table.
- Verify that the page does not break.
- Verify that Remediation column has 'N/A' for the recommendation with resolution:null
- Verify that rows for recommendations with no resolution can not be expanded

Once the ticket is verified, I will remove the temp commit before it PR gets merged.

### CODE REVIEW

File: src/SmartComponents/SystemAdvisor/SystemAdvisor.js [hunk 1]
```diff
@@ -285,14 +287,16 @@ const BaseSystemAdvisor = () => {
...
+        ? reportRow.filter((row) => row !== null)
+        : [];
     });
     //must recalculate parent for expandable table content whenever the array size changes
     builtRows.forEach((row, index) =>
       row.parent ? (row.parent = index - 1) : null
     );
 
-    systemAdvisorRef.current.rowCount = builtRows.length / 2;
+    systemAdvisorRef.current.rowCount = activeReports.length;
```
Comment (bastilian) [inline src/SmartComponents/SystemAdvisor/SystemAdvisor.js:299]: Why is this change needed?

**Review by gkarat — APPROVED** [review]

File: src/SmartComponents/SystemAdvisor/SystemAdvisor.js [hunk 2]
```diff
@@ -185,14 +185,14 @@ const BaseSystemAdvisor = () => {
...
       const selected = match?.selected;
       const isOpen = match?.isOpen || false;
-
       const reportRow = [
         {
           rule,
           resolution,
-          isOpen,
+          //make arrow button disappear when there is no resolution
+          isOpen: resolution ? isOpen : undefined,
```
[the remark this answers is not in the document]
Comment (mkholjuraev) [reply src/SmartComponents/SystemAdvisor/SystemAdvisor.js:193]: Nope, we should explecitly give undefined to force the arrow button disappear. If false is provided, arrow button will be visible but not in expanded state.

**Review by bastilian — APPROVED** [review]
LGTM! Thank you @mkholjuraev!

File: src/SmartComponents/SystemAdvisor/SystemAdvisor.js [hunk 3]
```diff
@@ -82,11 +82,11 @@ const BaseSystemAdvisor = () => {
   const hideResultsSatelliteManaged = !satelliteShowHosts && satelliteManaged;
   const getSelectedItems = (rows) => rows.filter((entity) => entity.selected);
   const selectedAnsibleRules = getSelectedItems(rows).filter(
-    (r) => r.resolution && r.resolution.has_playbook
+    (r) => r.resolution?.has_playbook
   );
   const selectedItemsLength = getSelectedItems(rows).length;
   const selectableItemsLength = rows.filter(
-    (r) => r.resolution && r.resolution.has_playbook
+    (r) => r.resolution?.has_playbook
```
[the remark this answers is not in the document]
Comment (mkholjuraev) [reply src/SmartComponents/SystemAdvisor/SystemAdvisor.js:89]: I would agree with @bastilian's suggestion as it looks more cleaner to me. In my opinion, we do not have to worry if the result will be false/undefined as both are handled in the same way in filter function.

[the remark this answers is not in the document]
Comment (gkarat) [reply src/SmartComponents/SystemAdvisor/SystemAdvisor.js:89]: OK :) this makes sense
